### ISSUE TITLE
Copy and Paste is Unrealiable in Both Grid-View and Text-View

### ISSUE THREAD

**Issue by Lonniebiz** [serge-rider/dbeaver#204]
I love Dbeaver 3.59 (Windows 10), but have an issue to report.

I'm have difficulty consistently copying data out of the grid in the bottom pane of the Dbeaver GUI.

Sometimes I can "right-click, copy, and paste grid data elsewhere" and some times I cannot. When this issue occurs with the grid, (at the same time) I have no issue copying and pasting text from the SQL Editor (in the top pane of dbeaver).

Additionally, you know how you can change the bottom grid from grid-view to text-view? Well, even when I do this, I have difficulty copying the text-view data from the bottom pane. And again, when I'm having this difficulty, I can immediately copy and paste text from the top-pane's SQL Editor without a problem. Its weird that the top pane never fails with copying and pasting but the bottom pane fails frequently.

This issue is sure hurting my workflow today. I'm working on a tough query where (each time I alter it) I'm having to compare "current outputs to previous outputs" closely using this online tool:
https://www.diffnow.com/

It is so frustrating that copying data from the grid is working only 1 out of 5 tries (even in text-view). I hope the developers can imagine why this could be happening and somehow ensure that this feature becomes extremely reliable.

**Comment by Lonniebiz** [comment]
I've been playing around with this issue a little more. I haven't quite figured out a pattern to it, but I think it has something to do with switching focus between the top and bottom panes. Sometimes, when this issue is occurring, if click in the top-pane (and therefore un-highlight any highlighted text in the top-pane, then click back into the bottom pane and select the table-data, sometimes this makes copy and pasting grid-data successful again.

The inability to copy and paste grid-data seems to have something to do with focus-changes between the top and bottom pane. Or maybe the issue has to do with data being highlighted in both panes at the same time. I haven't figured out a consistent pattern to the issue, but I'm hoping that these speculations may give the developers a lead towards reproducing this issue.

**Comment by serge-rider** [comment]
I can't reproduce that. Copy/paste are truly one of the most frequent operations with grid and I never saw any problem with them. So I think problem is in environment or in the way you are working with results.
- What is your OS?
- How do you perform copy/paste commands (context menu, keyboard shortcuts)?
- What actually happens - copy/paste commands are disabled, shortcuts do not work, ..?

**Comment by Lonniebiz** [comment]
Ok, granted, my set up is not normal.

I using Remmina's RDP client on Ubuntu Linux 16.04 to remote into a Window 10 hyper-v virtual machine over VPN where dbeaver is installed.

I realize there's a lot that could go wrong, but why is it that there is no issue copying and pasting from the top-pane of dbeaver? The issue only occurs in the bottom pane with grid-view-data and text-view-data.  The data I'm copying is only one column of 326 rows with each cell containing a 4 digit number (so it is not the size of the data causing this issue).

Plus, with this same setup I have, I'm able to copy and paste without issue in all other Windows 10 applications. The only place I encounter this issue is in the bottom-pane of dbeaver -- when copying and pasting grid-view-data and text-view-data.

So, even though my setup may seem problematic, the issue seems to be isolated to dbeaver's bottom pane.

**Comment by serge-rider** [comment]
Yep, it won't be easy to reproduce that.
Data grid is a sophisticated custom UI control, it heavily relies on focus gain/lost events. If something goes wrong with them then it may misfunction in some cases.

Could you still answer my questions above? And another couple of questions:
- Did you try to work with table data editor (without SQL pane). Is this problem happens there too?
- What if you hide SQL pane (F6) - does it solve focus problems?

**Comment by serge-rider** [comment]
Please check this issue still happens in 3.6

**Comment by Lonniebiz** [comment]
Sure. I made a quick video for you to see the behavior. It seems better, but still happens:
https://youtu.be/5l1bTfofiqA

Hope this helps.

**Comment by masebu** [comment]
Hi,

Having used dbeaver for a few months, I can't remember any issue with copy/paste until recently.
I've had trouble on 3.7.3 and 3.7.4
System-wise, I'm on Windows 10 build 1607. Local install, no remote desktop.

Same as OP, copying from SQL Editor and pasting into a different program (notepad++ for instance) is ok.
Copying from result set grid doesn't work, it doesn't update the clipboard: when I paste, it pastes my previous copy.

I checked the copy shortcut, I believe it's correctly defined. I checked with the menu: Copy is disabled.
<img width="213" alt="dbeaver_copy_result_unavailable" src="https://cloud.githubusercontent.com/assets/19706602/18632726/389c1658-7e79-11e6-974e-e052b230813b.png">


It occurs when I've used multi-tab results:
1. execute 1st statement as "Execute SQL statement": copy from grid "Result" is ok
2. execute 2nd statement as "Execute New SQL Statement" : copy from both grids "Result" and "Result - 2" is ok
3. execute 3rd statement as "Execute SQL statement": "Result - 2" grid disappers and "Result" grid is updated, which is normal; however copy command is now unavailable.

Any subsequent execution "Execute SQL statement": copy is still unavailable.
If I switch to other tabs of the lower pane (Explain Plan, Execution Log, Output) and back to Result: copy is again available.
Double-clicking on result tab itself doesn't change availability of the copy command.

I can reproduce it anytime with multi-tab results, but it may occur in other circumstances that I have not isolated yet.

Thanks a lot for looking into this, it is indeed quite disturbing.
